### ISSUE TITLE
coo2csr not working correctly for matrices where nnz < num_rows

### ISSUE THREAD

**Issue by jpola** [clMathLibraries/clSPARSE#153]
I found a bug related to my indices_to_offsets function. If number of nonzeros < number of rows the transformation function gives wrong row_offsets. row_offsets in this situation is larger than row_indices in corresponding coo format.

I missed this due to fact that for large matrices this situation is not common, but for small matrices I found that very often.

**Comment by jpola** [comment]
I have solution for that. Fix require scatter operation which I don't want to write since we will be using boost compute in the near future? Isn't it?

**Comment by kknox** [comment]
Yes, I've not heard significant resistance to using boost.compute; it makes a lot of sense for clsparse.
